roxie/ccd/ccdprotocol.cpp

```diff
@@ -1803,18 +1795,11 @@ class RoxieSocketWorker : public ProtocolQueryWorker
             {
                 mlResponseFmt = httpHelper.queryResponseMlFormat();
                 mlRequestFmt = httpHelper.queryRequestMlFormat();
-
-                StringAttr globalIdHeader, callerIdHeader;
-                const char *globalId = queryRequestGlobalIdHeader(httpHelper, logctx, globalIdHeader);
-                const char *callerId = queryRequestCallerIdHeader(httpHelper, logctx, callerIdHeader);
-                logctx.setHttpIdHeaderNames(globalIdHeader, callerIdHeader);
-                if (globalId && *globalId)
-                    msgctx->setTransactionId(globalId, callerId, true);  //logged and forwarded through SOAPCALL/HTTPCALL
-                else if (callerId && *callerId)
-                    msgctx->setCallerId(callerId); //may not matter, but maintain old behavior
             }
         }
 
+        msgctx->startSpan(nullptr, httpHelper.queryRequestHeaders());
```

Comment:
Spans will be used to analyze timings.  Since there is a delay from when the socket was accepted and when the span is created I think we should have saved off the real start times and pass them in here to be used for the span.

From the open telemetry c++ sdk docs:

struct StartSpanOptions
{
  // Optionally sets the start time of a Span.
  //
  // If the start time of a Span is set, timestamps from both the system clock
  // and steady clock must be provided.
  //
  // Timestamps from the steady clock can be used to most accurately measure a
  // Span's duration, while timestamps from the system clock can be used to most
  // accurately place a Span's
  // time point relative to other Spans collected across a distributed system.
  common::SystemTimestamp start_system_time;
  common::SteadyTimestamp start_steady_time;

Reply:
Created HPCC-30374 and pasted your comment from here in there.